### ISSUE TITLE
Ansible operator: skip execution on init

### ISSUE THREAD

**Issue by kvaps** [operator-framework/operator-sdk#1679]
## Feature Request

**Problem**

Every task execution is taking about 3s, even just simple playbook:

```
- hosts: localhost
  gather_facts: no
  tasks:
  - import_role:
      name: "memcached"
```

Eg if you have 100 custom resources, and then you restart operator, it will be able to process new operations only after 5 minutes, after processing all 100 existing resources.

**Solution**

Allow specifying `preserveStatus: true` option on a par with `reconcilePeriod: 0` in watches.yaml file:

```
- version: v1alpha1
  group: baz.example.com
  kind: Baz
  playbook: /opt/ansible/baz.yml
  reconcilePeriod: 0
  preserveStatus: true
```

*Processing:*

If `preserveStatus` is set to `true`, then save `metadata.generation` to `status.generation` for each resource.

*Initialization:*

During start do the check:  if  `metadata.generation` equals `status.generation`, then skip resource processing.

**Comment by shawn-hurley** [comment]
You could effectively get this now, by turning of dependent watches https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/dev/dependent_watches.md

WDYT?

**Comment by kvaps** [comment]
Hi @shawn-hurley, thanks for answer, but I'm not sure that this option can help somehow in this case.
Eg if I have 100 resources, ansible-operator will process all of them during the restart of the operator itself.

**Comment by kvaps** [comment]
Working prototype:

```yaml
  - meta: end_play
    when: "status.generation is defined and status.generation|int == metadata.generation|int"

  - debug:
      msg: "do something"

  - k8s_status:
      api_version: 'moonshot.wedos.org/v1alpha1'
      kind: 'MoonshotNode'
      name: '{{ metadata.name }}'
      status:
        generation: "{{ metadata.generation }}"
```

**Comment by kvaps** [comment]
But I don't know for sure how to get `.metadata.generation` and `.status` working from ansible-operator

**Comment by fabianvf** [comment]
I think this is likely not something we'll add, because there's no guarantee that the last reconciliation finished/succeeded or that cluster state has remained static since the operator restarted, so we want to ensure we reconcile on start. This may be a good example of where a hybrid operator could be the right pattern, since you could just override the reconcile logic to take resourceVersion into account and then reuse everything else. 

That being said the overhead is definitely a problem, bumping up the number of parallel jobs (as raised in #1678) and disabling fact gathering (as raised in #1677) should help bring that down, but if the performance is still unsatisfactory we should definitely focus some time on profiling/improving that.

**Comment by kvaps** [comment]
Maybe can we prioritize it somehow, for put dropping events above than initial tasks and reconciling?

**Comment by kvaps** [comment]
/remove-lifecycle stale
Sholud be solved by implementing watch bookmarks support https://github.com/operator-framework/operator-sdk/issues/1939

**Comment by fabianvf** [comment]
Watch bookmarks do look promising, if we can do this in a supported way then I'm all for it

**Comment by estroz** [comment]
Closing in favor of #1939, which solves the issue described here in another manner.

**Comment by mhrivnak** [comment]
I think should be re-opened.

Bookmarks are only useful when re-establishing a watch. That would not be the case when the operator container is first starting up. On startup, it always establishes new watches, because nothing is persisting a collection's resource version.

https://kubernetes.io/docs/reference/using-api/api-concepts/#watch-bookmarks

Further, bookmarks are only useful when you are watching with a label selector, which I don't think is supported yet by the sdk anyway.
